gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/AndroidPluginIntegration.kt

```diff
@@ -81,36 +81,32 @@ object AndroidPluginIntegration {
         val kaptProvider: TaskProvider<Task>? =
             project.locateTask(kotlinCompilation.compileTaskProvider.kaptTaskName)
 
-        val useLegacyApi = project.useLegacyVariantApi()
-        if (useLegacyApi) {
-            val sources = kotlinCompilation.androidVariant.getSourceFolders(SourceKind.JAVA)
-            kspTaskProvider.configure { task ->
-                // this is workaround for KAPT generator that prevents circular dependency
-                val filteredSources = Callable {
-                    val destinationProperty = (kaptProvider?.get() as? KaptTask)?.destinationDir
-                    val dir = destinationProperty?.get()?.asFile
-                    sources.filter { dir?.isParentOf(it.dir) != true }
-                }
-                when (task) {
-                    is KspTaskJvm -> { task.source(filteredSources) }
-                    is KspAATask -> { task.kspConfig.javaSourceRoots.from(filteredSources) }
-                    else -> Unit
-                }
+        val androidVariant = kotlinCompilation.androidVariant
+        if (androidVariant == null) {
+            throw RuntimeException(
+                "KSP is not compatible with built-in Kotlin. " +
```

Comment:
I guess what you mean by "built-in Kotlin" is AGP's built-in Kotlin. If so, should this be "KSP is not compatible with Android Gradle Plugin's built-in Kotlin. Please disable it by ..."?

Also, does a developer need to apply the kotlin plugin when `android.builtInKotlin=false` is set?